services-api/src/main/java/io/scalecube/services/ServiceCall.java

```diff
@@ -45,6 +45,7 @@
       req -> {
         // no-op
       };
+  private Map<String, String> credentials;
```

Comment:
Plz assign this map to some `Collections.emptyMap()`.  Having this would simply remove the need in changes in ServiceMessage: 

```
    /**
     * Adds to service message each element of given <code>headers</code>.
     *
     * @param headers given headers
     * @return self
     */
    public Builder headers(Map<String, String> headers) {
      if (headers != null) {
        headers.forEach(this::header);
      }
      return this;
    }
```

Reply:
Done https://github.com/scalecube/scalecube-services/pull/624/commits/72c78acfe6d73bb7d213c3cb2b6d0435a5715251